### PR TITLE
Semantic highlights in session based TSServer

### LINKED ISSUE #41262: Difference in behavior with encodedSemanticClassifications-full in TS 4.1 compared to using VS Code plugin
**TypeScript Version:**  4.1.0-dev.20201026

**Repro**
1. Using TS `4.1.0-dev.20201026` and a build of VS Code that explicitly disables the `typescript-vscode-sh-plugin` plugin

    Here's the change in the VS Code repo that disables the plugin:

    ```diff
    diff --git a/extensions/typescript-language-features/package.json b/extensions/typescript-language-features/package.json
    index dfd3f6295a..aa3763aaf7 100644
    --- a/extensions/typescript-language-features/package.json
    +++ b/extensions/typescript-language-features/package.json
    @@ -1114,12 +1114,6 @@
            }
            ]
        }
    -    ],
    -    "typescriptServerPlugins": [
    -      {
    -        "name": "typescript-vscode-sh-plugin",
    -        "enableForWorkspaceTypeScriptVersions": true
    -      }
        ]
    }
    }
    ```

1. Use the TS File:

    ```ts
    class Foo {}

    const a = Foo;

    console.log(a);
    ```

1. Look at semantic highlighting

**Bug**
With TS 4.1 and the plugin disabled, I see this response:

```
[Trace  - 00:50:46.802] <semantic> Response received: encodedSemanticClassifications-full (17). Request took 1673 ms. Success: true 
Result: {
    "spans": [
        6,
        3,
        11,
        24,
        3,
        11
    ],
    "endOfLineState": 0
}
```

With the plugin enabled, I see:

```
[Trace  - 00:50:59.813] <semantic> Response received: encodedSemanticClassifications-full (16). Request took 980 ms. Success: true 
Result: {
    "spans": [
        6,
        3,
        257,
        20,
        1,
        257,
        24,
        3,
        256,
        30,
        7,
        2064,
        38,
        3,
        3088,
        42,
        1,
        256
    ],
    "endOfLineState": 0
}
```

As you can see, the plugin seems to be providing a lot more information

@orta Is there any work we need to do on the VS Code side to adopt the new semantic work in TS 4.1?

/cc @aeschli

### PR DESCRIPTION
**Pull request by orta**
Fixes #41262 - I've cloned https://github.com/microsoft/vscode/tree/dev/mjbvz/semantic-ts locally and get working results from the tokenized files. I'm pretty sure this should be it @mjbvz! 

Screen Shot 2021-01-21 at 1 51 17 PM

### CODE REVIEW

File: src/services/shims.ts [hunk 1]
```diff
@@ -693,10 +693,10 @@ namespace ts {
             return realizeDiagnostics(diagnostics, newLine);
         }
 
-        public getSyntacticClassifications(fileName: string, start: number, length: number): string {
+        public getSyntacticClassifications(fileName: string, start: number, length: number, format?: SemanticClassificationFormat): string {
```
Comment (sheetalkamat) [inline src/services/shims.ts:696]: why is SemanticClassificationFormat sent to SyntacticClassifications ?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: src/services/shims.ts (hunk 1 above)
Comment (orta) [reply src/services/shims.ts]: Great point Sheetal, Thanks! I was fidgeting around and left this in by accident - the only changes needed are now in the PR

**Comment by orta** [comment]
@typescript-bot pack this

**Review by sheetalkamat — APPROVED** [review]

**Comment by orta** [comment]
Thanks Sheetal, it's a small change I'm pretty confident with - so I'm merging

**Pull request merged**